activerecord/test/cases/inheritance_test.rb

```diff
@@ -351,6 +351,9 @@ def test_inheritance_without_mapping
     assert_nothing_raised { s = SpecialSubscriber.new("name" => "And breaaaaathe!"); s.id = 'roger'; s.save }
   end
 
+  # NOTE: this test only cares that `Client` responds to `of_first_firm`,
+  # indicating that it inherited the scope from `Company`. It does not care
+  # if the scope and its joins and conditions are valid.
   def test_scope_inherited_properly
     assert_nothing_raised { Company.of_first_firm }
     assert_nothing_raised { Client.of_first_firm }
```

Comment:
@rafaelsales depends on how `scope` is implemented if it's more involved than a class method which we'd want to test for regressions, but I agree with the sentiment.

A quick look at https://github.com/rails/rails/blob/428d47adfed8d6aa7b21aec2bf5ad890961c9de3/activerecord/lib/active_record/scoping/named.rb#L164 shows that it's indeed a class method defined on the `singleton_class`, so we're basically just testing Ruby's inheritance here.